### PR TITLE
feat(ban): Add rule to ban shell execution via backticks

### PR DESCRIPTION
**Pull request by BitScout**
Resolves https://github.com/ekino/phpstan-banned-code/issues/22

Bans use of shell execution by putting a command in backticks: https://www.php.net/manual/en/language.operators.execution.php

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,11 @@ master
 
 * todo...
 
+v1.1.0
```
Comment (mremi) [inline CHANGELOG.md:9]: only master for now

File: snippets/backticks.php [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<?php
+
+``;
```
Comment (mremi) [inline snippets/backticks.php:3]:
```suggestion
`ls -lsa`;
```

File: snippets/proc_open.php [hunk 3]
```diff
@@ -1,4 +1,3 @@
 <?php
 $pipes = [];
```
Comment (mremi) [inline snippets/proc_open.php:2]: please add blank line before :)

File: snippets/backticks.php (hunk 2 above)
Comment (BitScout) [reply snippets/backticks.php:3]: I thought about that, but the `exec` example is empty too and I checked, the violation is reported even if the backticks  have no content.

Comment (mremi) [reply snippets/backticks.php:3]: please update exec as well :)

File: tests/Rules/BannedNodesRuleTest.php [hunk 4]
```diff
@@ -134,5 +136,6 @@ public function getHandledNodes(): \Generator
     {
         yield [new Eval_($this->createMock(Expr::class))];
         yield [new Exit_()];
+        yield [new ShellExec([''])];
```
Comment (mremi) [inline tests/Rules/BannedNodesRuleTest.php:139]: phpdoc is not true anymore :) `\Generator<array<mixed>>`

**Pull request merged**